scripts/utils.ts

```diff
@@ -0,0 +1,63 @@
+import { NetworkId } from './types'
+import { registryContractAbi } from '../abis/Registry'
+import { mainnet, arbitrum, optimism, polygon, base, celo } from 'viem/chains'
+import {
+  createPublicClient,
+  http,
+  getContract,
+  Address,
+  PublicClient,
+} from 'viem'
+
+const NETWORK_ID_TO_VIEM_CLIENT = {
+  [NetworkId['ethereum-mainnet']]: createPublicClient({
+    chain: mainnet,
+    transport: http(),
+  }),
+  [NetworkId['arbitrum-one']]: createPublicClient({
+    chain: arbitrum,
+    transport: http(),
+  }),
+  [NetworkId['op-mainnet']]: createPublicClient({
+    chain: optimism,
+    transport: http(),
+  }),
+  [NetworkId['celo-mainnet']]: createPublicClient({
+    chain: celo,
+    transport: http(),
+  }),
+  [NetworkId['polygon-pos-mainnet']]: createPublicClient({
+    chain: polygon,
+    transport: http(),
+  }),
+  [NetworkId['base-mainnet']]: createPublicClient({
+    chain: base,
+    transport: http(),
+  }),
+} as unknown as Partial<Record<NetworkId, PublicClient>>
+
+/**
+ * Gets a public Viem client for a given NetworkId
+ */
+export function getViemPublicClient(networkId: NetworkId) {
+  const client = NETWORK_ID_TO_VIEM_CLIENT[networkId]
+  if (!client) {
+    throw new Error(`No viem client found for networkId: ${networkId}`)
+  }
+  return client
+}
+
+/**
+ * Returns a contract object representing the registry
+ */
+export async function getRegistryContract(
```

Comment:
nit: This could just accept a `networkId` and do the translation internally, since there's only going to be one registry per network.